src/Reflection/Php/PhpFunctionFromParserNodeReflection.php

```diff
@@ -281,7 +281,10 @@ private function nodeIsOrContainsYield(Node $node): bool
 		foreach ($node->getSubNodeNames() as $nodeName) {
 			$nodeProperty = $node->$nodeName;
 
-			if ($nodeProperty instanceof Node && $this->nodeIsOrContainsYield($nodeProperty)) {
+			if ($nodeProperty instanceof Node &&
+				!$nodeProperty instanceof FunctionLike &&
```

Comment:
Would be great to also not descend into anonymous classes.

Reply:
@ondrejmirtes Anonymous class nodes should now be skipped as well. I submitted the fix via Roave/BetterReflection#1479.